SampleApp/BoxView.swift

```diff
@@ -9,31 +9,155 @@ import SwiftUI
 import Warp
 
 struct BoxView: View {
+    @State private var style: Warp.BoxStyle = .info
+
+    @State private var shouldShowIcon = true
+
+    @State private var boxTitle = "Title"
+
+    @State private var boxSubtitle = "Use this variant to call extra attention to useful, contextual information."
+
+    @State private var hasLink = false
+
+    @State private var hasButton = false
+
+    @Environment(\.colorScheme) private var colorScheme
+
     var body: some View {
-        ScrollView(showsIndicators: false) {
-            ForEach(Warp.BoxStyle.allCases, id: \.self) { style in
-                Warp.Box(
-                    style: style,
-                    title: "Title",
```

Comment:
We don't have this variant, so if title is nil the icon should hide also. Probably this is the change we need to make. Otherwise the demo app looks great.
![Simulator Screenshot - iPhone 15 - 2024-02-13 at 19 57 24](https://github.com/warp-ds/warp-ios/assets/89784012/cbf53366-bf01-4ab9-886c-4e247711d49a)
<img width="667" alt="Screenshot 2024-02-13 at 20 00 18" src="https://github.com/warp-ds/warp-ios/assets/89784012/56591c4c-21e7-43b8-97f9-d8c10db08726">

Reply:
oh no the logic for hiding title when it is empty is in the demoApp :D 
We should move it to the Warp component itself, otherwise it won't work for the users.
<img width="772" alt="Screenshot 2024-02-13 at 20 07 16" src="https://github.com/warp-ds/warp-ios/assets/89784012/b6c274f7-e0d4-4fbd-a9d3-10b96493282d">
<img width="757" alt="Screenshot 2024-02-13 at 20 06 13" src="https://github.com/warp-ds/warp-ios/assets/89784012/2c3b1998-9c88-4b77-8749-aaebdc1311af">

Reply:
Great catch, thanks!
Updated.